application/src/main/java/org/togetherjava/tjbot/commands/moderation/ReportCommand.java

```diff
@@ -0,0 +1,196 @@
+package org.togetherjava.tjbot.commands.moderation;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import net.dv8tion.jda.api.EmbedBuilder;
+import net.dv8tion.jda.api.JDA;
+import net.dv8tion.jda.api.entities.Message;
+import net.dv8tion.jda.api.entities.MessageEmbed;
+import net.dv8tion.jda.api.entities.channel.concrete.TextChannel;
+import net.dv8tion.jda.api.events.interaction.ModalInteractionEvent;
+import net.dv8tion.jda.api.events.interaction.command.MessageContextInteractionEvent;
+import net.dv8tion.jda.api.interactions.InteractionHook;
+import net.dv8tion.jda.api.interactions.commands.build.Commands;
+import net.dv8tion.jda.api.interactions.components.Modal;
+import net.dv8tion.jda.api.interactions.components.text.TextInput;
+import net.dv8tion.jda.api.interactions.components.text.TextInputStyle;
+import net.dv8tion.jda.api.requests.restaction.MessageCreateAction;
+import net.dv8tion.jda.api.utils.Result;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.togetherjava.tjbot.commands.BotCommandAdapter;
+import org.togetherjava.tjbot.commands.CommandVisibility;
+import org.togetherjava.tjbot.commands.MessageContextCommand;
+import org.togetherjava.tjbot.config.Config;
+
+import java.awt.*;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Predicate;
+import java.util.regex.Pattern;
+
+
+/**
+ * Implements the /report command, which allows users to contact a moderator within the server which
+ * forwards messages to moderators in a dedicated channel given by
+ * {@link Config#getModMailChannelPattern()}.
+ *
+ * The /report command would give us necessary information about the offending user, and specific
+ * message that was offensive.
```

Comment:
"us". javadoc is written with the coder as author, not the moderator of the server.

maybe it would also be better to merge this text with the first paragraph. maybe:

> Implements the /report command, which allows users to report a selected offensive message from another user.
> The message is then forwarded to moderators in a dedicated channel given by ...